<html><h3>:x: ci:test - 16 out of 18 jobs passed in 1 hour 39 minutes 42 seconds 40 ms</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/com-liferay-commerce/tree/7.1.x">7.1.x</a><br/>Branch GIT ID: <a href="https://github.com/liferay/com-liferay-commerce/commit/ee1ef5f493f5e3b131d446d0ac609caf3c87b9e6">ee1ef5f493f5e3b131d446d0ac609caf3c87b9e6</a></p>16 out of 18 jobs PASSED<div><h4>2 Failed Jobs:</h4><ul><li><a href="https://test-1-20.liferay.com/job/test-subrepository-acceptance-pullrequest(7.1.x)/90/"><strike><strong>test-subrepository-acceptance-pullrequest(7.1.x)</strong></strike></a></li><li><a href="https://test-1-23.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/455/"><strike><strong>test-subrepository-acceptance-pullrequest-batch(7.1.x)/subrepository-integration-mysql57-jdk8</strong></strike></a></li></ul></div><details><summary><strong>16 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-23.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/451/">test-subrepository-acceptance-pullrequest-batch(7.1.x)/functional-smoke-tomcat90-mysql57-jdk8/0</a></li><li><a href="https://test-1-5.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/412/">test-subrepository-acceptance-pullrequest-batch(7.1.x)/subrepository-assemble-jdk8</a></li><li><a href="https://test-1-22.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/546/">test-subrepository-acceptance-pullrequest-batch(7.1.x)/subrepository-functional-tomcat90-mysql57-jdk8/0</a></li><li><a href="https://test-1-23.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/452/">test-subrepository-acceptance-pullrequest-batch(7.1.x)/subrepository-functional-tomcat90-mysql57-jdk8/1</a></li><li><a href="https://test-1-22.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/549/">test-subrepository-acceptance-pullrequest-batch(7.1.x)/subrepository-functional-tomcat90-mysql57-jdk8/10</a></li><li><a href="https://test-1-5.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/414/">test-subrepository-acceptance-pullrequest-batch(7.1.x)/subrepository-functional-tomcat90-mysql57-jdk8/11</a></li><li><a href="https://test-1-21.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/335/">test-subrepository-acceptance-pullrequest-batch(7.1.x)/subrepository-functional-tomcat90-mysql57-jdk8/2</a></li><li><a href="https://test-1-22.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/547/">test-subrepository-acceptance-pullrequest-batch(7.1.x)/subrepository-functional-tomcat90-mysql57-jdk8/3</a></li><li><a href="https://test-1-21.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/336/">test-subrepository-acceptance-pullrequest-batch(7.1.x)/subrepository-functional-tomcat90-mysql57-jdk8/4</a></li><li><a href="https://test-1-23.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/453/">test-subrepository-acceptance-pullrequest-batch(7.1.x)/subrepository-functional-tomcat90-mysql57-jdk8/5</a></li><li><a href="https://test-1-5.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/413/">test-subrepository-acceptance-pullrequest-batch(7.1.x)/subrepository-functional-tomcat90-mysql57-jdk8/6</a></li><li><a href="https://test-1-22.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/548/">test-subrepository-acceptance-pullrequest-batch(7.1.x)/subrepository-functional-tomcat90-mysql57-jdk8/7</a></li><li><a href="https://test-1-23.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/454/">test-subrepository-acceptance-pullrequest-batch(7.1.x)/subrepository-functional-tomcat90-mysql57-jdk8/8</a></li><li><a href="https://test-1-21.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/337/">test-subrepository-acceptance-pullrequest-batch(7.1.x)/subrepository-functional-tomcat90-mysql57-jdk8/9</a></li><li><a href="https://test-1-22.liferay.com/job/test-subrepository-acceptance-pullrequest-dist(7.1.x)/69/">test-subrepository-acceptance-pullrequest-dist(7.1.x)/jdk8</a></li><li><a href="https://test-1-22.liferay.com/job/test-subrepository-acceptance-pullrequest-validation(7.1.x)/129/">test-subrepository-acceptance-pullrequest-validation(7.1.x)/jdk8</a></li></ul></details><h5>For more details click <a href="https://test-1-20.liferay.com/userContent/jobs/test-subrepository-acceptance-pullrequest(7.1.x)/builds/90/jenkins-report.html">here</a>.</h5><hr/><h4>Failures unique to this pull:</h4><ol><li><div><h5><a href="https://test-1-23.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/455/">test-subrepository-acceptance-pullrequest-batch(7.1.x)/subrepository-integration-mysql57-jdk8</a></h5><div><h6>Job Results:</h6><p>1805 Tests Passed.<br/>2 Tests Failed.</p></div><ol><li><div><a href="https://test-1-23.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/AXIS_VARIABLE=0,label_exp=!master/455//consoleText">AXIS_VARIABLE=0,label_exp=!master #455</a><ol><li><div><a href="https://test-1-23.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/AXIS_VARIABLE=0,label_exp=!master/455//testReport/com.liferay.commerce.payment.engine.test/CommercePaymentEngineTest/testCompletePayment">CommercePaymentEngineTest.testCompletePayment</a><pre><code>java.lang.AssertionError: expected:&lt;2&gt; but was:&lt;1&gt;
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.liferay.commerce.payment.engine.test.CommercePaymentEngineTest.testCompletePayment(CommercePaymentEngineTest.java:140)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.jboss.arquillian.junit.Arquillian$8$1.invoke(Arquillian.java:370)
	at org.jboss.arquillian.container.test.impl.execution.LocalTestExecuter.execute(LocalTestExecuter.java:60)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:94)
	at org.jboss.arquilli...</code></pre></div></li><li><div><a href="https://test-1-23.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/AXIS_VARIABLE=0,label_exp=!master/455//testReport/com.liferay.commerce.payment.engine.test/CommercePaymentEngineTest/testProcessPayment">CommercePaymentEngineTest.testProcessPayment</a><pre><code>java.lang.AssertionError: expected:&lt;2&gt; but was:&lt;1&gt;
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.liferay.commerce.payment.engine.test.CommercePaymentEngineTest.testProcessPayment(CommercePaymentEngineTest.java:202)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.jboss.arquillian.junit.Arquillian$8$1.invoke(Arquillian.java:370)
	at org.jboss.arquillian.container.test.impl.execution.LocalTestExecuter.execute(LocalTestExecuter.java:60)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:94)
	at org.jboss.arquillia...</code></pre></div></li></ol></div></li></ol></div></li></ol></details></html>

--- a/commerce-checkout-web/src/main/java/com/liferay/commerce/checkout/web/internal/util/PaymentProcessCommerceCheckoutStep.java
+++ b/commerce-checkout-web/src/main/java/com/liferay/commerce/checkout/web/internal/util/PaymentProcessCommerceCheckoutStep.java
@@ -19,7 +19,12 @@ import com.liferay.commerce.checkout.web.internal.display.context.PaymentProcess
 import com.liferay.commerce.checkout.web.util.BaseCommerceCheckoutStep;
 import com.liferay.commerce.checkout.web.util.CommerceCheckoutStep;
 import com.liferay.commerce.checkout.web.util.CommerceCheckoutStepServicesTracker;
+import com.liferay.commerce.constants.CommerceWebKeys;
+import com.liferay.commerce.context.CommerceContext;
+import com.liferay.commerce.currency.model.CommerceMoney;
 import com.liferay.commerce.model.CommerceOrder;
+import com.liferay.commerce.price.CommerceOrderPrice;
+import com.liferay.commerce.price.CommerceOrderPriceCalculation;
 import com.liferay.frontend.taglib.servlet.taglib.util.JSPRenderer;
 import com.liferay.portal.kernel.util.Portal;
 
@@ -65,7 +70,17 @@ public class PaymentProcessCommerceCheckoutStep
 			(CommerceOrder)httpServletRequest.getAttribute(
 				CommerceCheckoutWebKeys.COMMERCE_ORDER);
 
-		if (BigDecimal.ZERO.compareTo(commerceOrder.getTotal()) == 0) {
+		CommerceContext commerceContext =
+			(CommerceContext)httpServletRequest.getAttribute(
+				CommerceWebKeys.COMMERCE_CONTEXT);
+
+		CommerceOrderPrice commerceOrderPrice =
+			_commerceOrderPriceCalculation.getCommerceOrderPrice(
+				commerceOrder, commerceContext);
+
+		CommerceMoney orderPriceTotal = commerceOrderPrice.getTotal();
+
+		if (BigDecimal.ZERO.compareTo(orderPriceTotal.getPrice()) == 0) {
 			return false;
 		}
 
@@ -123,6 +138,9 @@ public class PaymentProcessCommerceCheckoutStep
 	@Reference
 	private CommerceCheckoutStepServicesTracker
 		_commerceCheckoutStepServicesTracker;
+
+	@Reference
+	private CommerceOrderPriceCalculation _commerceOrderPriceCalculation;
 
 	@Reference
 	private JSPRenderer _jspRenderer;

--- a/commerce-payment-method-paypal/src/main/java/com/liferay/commerce/payment/method/paypal/internal/servlet/CommercePaymentMethodPayPalServlet.java
+++ b/commerce-payment-method-paypal/src/main/java/com/liferay/commerce/payment/method/paypal/internal/servlet/CommercePaymentMethodPayPalServlet.java
@@ -91,18 +91,19 @@ public class CommercePaymentMethodPayPalServlet extends HttpServlet {
 					commerceOrder.getCommerceOrderId(), paymentId,
 					httpServletRequest);
 			}
-
-			String token = ParamUtil.getString(httpServletRequest, "token");
-
-			if (paymentId.isEmpty() && !token.isEmpty()) {
-				_commercePaymentEngine.completeRecurringPayment(
-					commerceOrder.getCommerceOrderId(), token,
-					httpServletRequest);
-			}
 			else {
-				_commercePaymentEngine.completePayment(
-					commerceOrder.getCommerceOrderId(), paymentId,
-					httpServletRequest);
+				String token = ParamUtil.getString(httpServletRequest, "token");
+
+				if (paymentId.isEmpty() && !token.isEmpty()) {
+					_commercePaymentEngine.completeRecurringPayment(
+						commerceOrder.getCommerceOrderId(), token,
+						httpServletRequest);
+				}
+				else {
+					_commercePaymentEngine.completePayment(
+						commerceOrder.getCommerceOrderId(), paymentId,
+						httpServletRequest);
+				}
 			}
 
 			String redirect = ParamUtil.getString(

